avocado/utils/cpu.py

```diff
@@ -37,13 +37,16 @@ def _list_matches(lst, pattern):
 
 def _get_cpu_info():
     """
-    Reads /proc/cpuinfo and returns a list of file lines
+    Returns info on the 1st CPU entry from /proc/cpuinfo as a list of lines
 
-    :returns: `list` of lines from /proc/cpuinfo file
+    :returns: `list` of lines 1st CPU entry from /proc/cpuinfo file
     :rtype: `list`
     """
-    with open('/proc/cpuinfo', 'r') as f:
-        cpuinfo = f.readlines()
+    cpuinfo = []
+    for line in open('/proc/cpuinfo').readlines():
+        if line == '\n':
+            break
+        cpuinfo.append(line)
```

Comment:
This is an **internal**, **private** function.  The only usages we should care about are the ones here, and they seem to covered.  If this brings breakages to what their current users (the public functions), then those breakages should be addressed (one way to address it is to NACK this change).

Also, I noticed the detail about the extra info on Power, and still, the usages here seem to be covered.

The docstring has already been updated to describe accurately what it would return.  We could go one step further, and change its name, but at the same time, one could argue that the function you are thinking of should be better named `_get_all_cpus_info()`.